src/clib/pioc_support.cpp

```diff
@@ -553,46 +553,115 @@ adios2_variable* spio_define_adios2_variable(iosystem_desc_t *ios, file_desc_t *
+                adiosErr = adios2_set_operation_parameter(variable, operation_index, "doshuffle", "BLOSC_BITSHUFFLE");
+                if (adiosErr != adios2_error_none)
+                {
+                    pio_err(ios, file, PIO_EADIOS2ERR, __FILE__, __LINE__,
+                            "Failed to set Blosc2 shuffle (doshuffle=BLOSC_BITSHUFFLE) for variable %s (adios2_error=%s)",
+                            name, convert_adios2_error_to_string(adiosErr));
+                }
+            }
+        }
+        else if (variable_compression_method == ADIOS_COMPRESSION_METHOD_BZIP2)
         {
-            pio_err(ios, file, PIO_EADIOS2ERR, __FILE__, __LINE__,
-                    "Failed to set Blosc2 shuffle (doshuffle=BLOSC_BITSHUFFLE) for variable %s (adios2_error=%s)",
-                    name, convert_adios2_error_to_string(adiosErr));
+            if (ios->lossless_compression_operator != NULL)
+            {
+                /* Add a compression operation (BZip2) to this variable */
+                adiosErr = adios2_add_operation(&operation_index, variable, ios->lossless_compression_operator, "", "");
+                if (adiosErr != adios2_error_none)
+                {
+                    pio_err(ios, file, PIO_EADIOS2ERR, __FILE__, __LINE__,
+                            "Failed to add BZip2 compression operation to variable %s (adios2_error=%s)",
+                            name, convert_adios2_error_to_string(adiosErr));
+                }
+            }
         }
-        #elif defined(ADIOS2_HAVE_BZIP2)
-        /* Add a compression operation (BZip2) to this variable */
-        adiosErr = adios2_add_operation(&operation_index, variable, ios->compression_operator, "", "");
-        if (adiosErr != adios2_error_none)
+        #ifdef _SPIO_ADIOS_USE_LOSSY_COMPRESSION
+        else if (variable_compression_method == ADIOS_COMPRESSION_METHOD_MGARD)
+        {
+            if (ios->lossy_compression_operator != NULL)
+            {
+                /* Add a lossy compression operation (MGARD) to this variable */
+                adiosErr = adios2_add_operation(&operation_index, variable, ios->lossy_compression_operator, "accuracy", "0.001");
+                if (adiosErr != adios2_error_none)
+                {
+                    pio_err(ios, file, PIO_EADIOS2ERR, __FILE__, __LINE__,
+                            "Failed to add MGARD compression operation to variable %s (adios2_error=%s)",
+                            name, convert_adios2_error_to_string(adiosErr));
+                }
+            }
+        }
+        else if (variable_compression_method == ADIOS_COMPRESSION_METHOD_SZ)
+        {
+            if (ios->lossy_compression_operator != NULL)
+            {
+                /* Add a lossy compression operation (SZ) to this variable */
+                adiosErr = adios2_add_operation(&operation_index, variable, ios->lossy_compression_operator, "accuracy", "0.001");
```

Comment:
Please add info on the error bound used in the configure messages for now. We need to allow users to configure these settings in future